java/xen-api-samples/pom.xml

```diff
@@ -47,9 +47,19 @@
     </properties>
     <dependencies>
         <dependency>
-            <groupId>com.citrix.hypervisor</groupId>
+            <groupId>com.xenserver</groupId>
             <artifactId>xen-api</artifactId>
-            <version>[8.1.0,)</version>
+            <version>[23.12.0,)</version>
+        </dependency>
+        <dependency>
```

Comment:
These are `xen-api` rather than `xen-api-samples` dependencies. Unless the code in here does something to access xmlrpc code directly (which would not be great) we shouldn't need to list them in this package's dependencies.

Reply:
Unfortunately they are necessary, doesn't compile without them. The compiler complains it can't acces some XMLRPC exception classes. It's odd that it didn't before, probably something to do with the fact that the JAR is compiled using the new JDK?

Reply:
xapi `23.12.0` is in RT next but only recently.

Reply:
Looks like the SDK commit didn't make it into 23.12.0, so this PR needs updating regardless. 

https://github.com/xapi-project/xen-api/releases/tag/untagged-0e0b85c1490d3fb138e8

We will first test a fully ready SDK before updating this PR, regardless. I'll set it into draft.

Reply:
Can confirm `xmlrpc-common` is needed using the new SDK:

The `Network.create` call in `AddNetwork.TestCore` can throw a `XmlRpcException`. This requires `xmlrpc-common`
![image](https://github.com/xenserver/xenserver-samples/assets/32554698/88821eba-ea5a-4278-826b-21311db458e0)

However `xmlrpc-client` isn't needed. I managed to run the tests without it just fine, so I'll remove it 👍

Reply:
Also, can confirm that xapi 23.12.0 and above supports the new SDK so we can use that as a minimum version number for these samples

Reply:
> Maybe the project wasn't cleaned of existing dependency JARs

Possibly, to be honest I didn't spend too much time on it to investigate.

> The Network.create call in AddNetwork.TestCore can throw a XmlRpcException

You're right, I had forgotten that in fact all calls in the Java SDK throw XmlRpcException. Which means the design of the SDK may need improvement, because theoretically its purpose is to hide the implementation details - the client using the SDK should not have to catch xcmlrpc exception, if we changed to the jsonrpc backend, for example, the client would have to change the code.